Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt

```diff
@@ -0,0 +1,95 @@
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.doNavigate
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class SettingsPrivacyPreservingAnalyticsFragment :
+    Fragment(R.layout.fragment_settings_privacy_preserving_analytics),
+    AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val vm: SettingsPrivacyPreservingAnalyticsViewModel by cwaViewModels { viewModelFactory }
+    private val binding: FragmentSettingsPrivacyPreservingAnalyticsBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.apply {
+
+            settingsPpaHeader.headerButtonBack.buttonIcon.setOnClickListener {
+                (activity as MainActivity).goBack()
+            }
+
+            settingsPpaSwitchRow.settingsSwitchRowSwitch.setOnCheckedChangeListener { view, _ ->
+                // Make sure that listener is called by user interaction
+                if (!view.isPressed) return@setOnCheckedChangeListener
+
+                vm.analyticsToggleEnabled()
+            }
+
+            federalStateRow.setOnClickListener {
+                doNavigate(
+                    SettingsPrivacyPreservingAnalyticsFragmentDirections
+                        .actionSettingsPrivacyPreservingAnalyticsFragmentToAnalyticsUserInputFragment(
+                        type = AnalyticsUserInputFragment.InputType.FEDERAL_STATE
+                    )
+                )
+            }
+
+            districtRow.setOnClickListener {
+                doNavigate(
+                    SettingsPrivacyPreservingAnalyticsFragmentDirections
+                        .actionSettingsPrivacyPreservingAnalyticsFragmentToAnalyticsUserInputFragment(
+                        type = AnalyticsUserInputFragment.InputType.DISTRICT
+                    )
+                )
+            }
+            ageGroupRow.setOnClickListener {
+                doNavigate(
+                    SettingsPrivacyPreservingAnalyticsFragmentDirections
+                        .actionSettingsPrivacyPreservingAnalyticsFragmentToAnalyticsUserInputFragment(
+                        type = AnalyticsUserInputFragment.InputType.AGE_GROUP
+                    )
+                )
+            }
+        }
+
+        vm.ageGroup.observe2(this) {
+            binding.ageGroupRowBody.text = getString(it.labelStringRes)
+        }
+
+        vm.federalState.observe2(this) {
+            binding.districtRow.visibility = if (it != PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED) {
+                View.VISIBLE
+            } else {
+                View.GONE
+            }
```

Comment:
binding.districtRow.isVisible = it != PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED